tests/dtypes/test_money.py

```diff
@@ -0,0 +1,128 @@
+from typing import cast
+
+from pytest import FixtureRequest
+
+from spinta.backends.postgresql.components import PostgreSQL
+from spinta.commands.write import dataitem_from_payload
+from spinta.core.config import RawConfig
+from spinta.testing.client import create_test_client
+from spinta.testing.data import listdata
+from spinta.testing.manifest import bootstrap_manifest
+
+import pytest
+
+from spinta import commands
+from spinta.manifests.components import Manifest
+from spinta.components import Store
+
+
+def test_money(
+    rc: RawConfig,
+    postgresql: str,
+    request: FixtureRequest,
+):
+    context = bootstrap_manifest(rc, '''
+    d | r | b | m | property                   | type           | ref
+    backends/postgres/dtypes/money             |                |
+      |   |   | City                           |                |
+      |   |   |   | name                       | string         |
+      |   |   |   | amount                     | money          | EUR
+    ''', backend=postgresql, request=request)
+
+    model: str = 'backends/postgres/dtypes/money/City'
+
+    app = create_test_client(context)
+    app.authmodel(model, [
+        'insert',
+        'getall',
+    ])
+
+    # Write data
+    resp = app.post(f'/{model}', json={
+        'name': "Vilnius",
+        'amount': 100
+    })
+    assert resp.status_code == 201
+
+    # Read data
+    resp = app.get(f'/{model}')
+    assert listdata(resp, full=True) == [
+        {
+            'name': "Vilnius",
+            'amount': 100
+        }
+    ]
+
+
+def test_money_currency(
+    rc: RawConfig,
+    postgresql: str,
+    request: FixtureRequest,
+):
+    context = bootstrap_manifest(rc, '''
+    d | r | b | m | property                   | type           | ref
+    backends/postgres/dtypes/money             |                |
+      |   |   | City                           |                |
+      |   |   |   | name                       | string         |
+      |   |   |   | amount                     | money          | 
+      |   |   |   | currency                   | string         | 
+    ''', backend=postgresql, request=request)
+
+    model: str = 'backends/postgres/dtypes/money/City'
+
+    app = create_test_client(context)
+    app.authmodel(model, [
+        'insert',
+        'getall',
+    ])
+
+    # Write data
+    resp = app.post(f'/{model}', json={
+        'name': "Vilnius",
+        'amount': 100,
```

Comment:
If `ref` is not given, then  full form (`{'_amount': 100, '_currency': 'EUR'}`) should be given.

Reply:
```
E   sqlalchemy.exc.ProgrammingError: (psycopg2.ProgrammingError) can't adapt type 'dict'
E   [SQL: INSERT INTO "backends/postgres/dtypes/money/City" (_txn, _created, _id, _revision, name, amount, "amount._currency") VALUES (%(_txn)s, TIMEZONE('utc', CURRENT_TIMESTAMP), %(_id)s, %(_revision)s, %(name)s, %(amount)s, %(amount._currency)s)]
E   [parameters: {'_txn': UUID('2097700c-9c33-4e55-ab02-1df5ee0dc3de'), '_id': UUID('ab91da3f-92a9-4720-bd3a-5a4dfd2ac0f3'), '_revision': 'a7d2953c-897e-42d0-868a-780626ef790b', 'name': 'Vilnius', 'amount': {'_amount': 100, '_currency': 'EUR'}, 'amount._currency': None}]
E   (Background on this error at: https://sqlalche.me/e/14/f405)
```

Reply:
For write:
spinta/backends/postgesql/commands/write.py def insert() -> patch (money, money._currency, money._amount)